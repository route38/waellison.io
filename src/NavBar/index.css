#navBar {
    width: 100%;
    grid-column-start: 1;
    grid-column-end: 6;
    grid-row-start: 1;
    grid-row-end: 2;
    padding: 0.5rem 2rem;
    box-sizing: border-box;
    color: var(--foreground-dark);
    background-color: var(--background-black);
    display: flex;
    align-items: center;
}

#navBar nav {
	display: flex;
	flex-direction: row;
	flex-wrap: nowrap;
	justify-content: space-between;
    width: 100%;
}

#topLevelNavMenu li {
    list-style-type: none;
    padding: 0 0.5rem;
    display: inline-block;
}

.navLink:link, .navLink:visited {
  color: white;
  text-decoration: none;  
}

.navLink:hover {
  color: #ec7600;
  border-bottom: 1px solid #7600ec;
}

#navBarBody menu {
  margin: 0;
  padding: 0;
}

#navBarBrand .navLink:link, #navbarBrand .navLink:visited {
  background-image: linear-gradient(var(--ua-azure) 50%, var(--ua-yellow) 50%);
  background-clip: text;
  color: transparent;
  font-weight: 900;
  text-transform: uppercase;
}
