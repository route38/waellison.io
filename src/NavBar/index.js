import React from "react";
import "./index.css";

const NavBar = () => {
    return (
        <div id="navBar">
            <nav>
                <div id='navBarBrand'>
                    <a className="navLink" title="Україні Слава!" href="/">
                        William Ellison
                    </a>
                </div>
                <div id='navBarBody'>
                    <menu id="topLevelNavMenu">
                        <li>
                            <a className="navLink" href="#resume">
                                Résumé
                            </a>
                        </li>
                        <li>
                            <a className="navLink" href="#portfolio">
                                Portfolio
                            </a>
                        </li>
                    </menu>
                </div>
            </nav>
        </div>
    )
}

export default NavBar;