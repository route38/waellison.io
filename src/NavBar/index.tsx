import React from "react";
import { Link } from "react-router-dom";
import "./index.css";

const NavBar = () => {
  return (
    <div id="navBar">
      <nav>
        <div id="navBarBrand">
          <a className="navLink" title="Україні Слава!" href="/">
            William Ellison
          </a>
        </div>
        <div id="navBarBody">
          <menu id="topLevelNavMenu">
            <li>
              <a className="navLink neverGonnaGiveYouUp" href="//waellison.blog">
                Blog
              </a>
            </li>
            <li>
                <Link className="navLink neverGonnaLetYouDown" to="/resume">Résumé</Link>
            </li>
            <li>
                <Link className="navLink neverGonnaRunAroundAndDesertYou" to="/portfolio">Portfolio</Link>
            </li>
          </menu>
        </div>
      </nav>
    </div>
  );
};

export default NavBar;
