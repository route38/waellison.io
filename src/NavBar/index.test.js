import React from "react";
import renderer from "react-test-renderer";
import NavBar from './index';

it('renders correct navbar component', () => {
    const component = renderer.create(<NavBar />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
