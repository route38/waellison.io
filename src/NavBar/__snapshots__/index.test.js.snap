// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct navbar component 1`] = `
<div
  id="navBar"
>
  <nav>
    <div
      id="navBarBrand"
    >
      <a
        className="navLink"
        href="/"
        title="Україні Слава!"
      >
        William Ellison
      </a>
    </div>
    <div
      id="navBarBody"
    >
      <menu
        id="topLevelNavMenu"
      >
        <li>
          <a
            className="navLink"
            href="#resume"
          >
            Résumé
          </a>
        </li>
        <li>
          <a
            className="navLink"
            href="#portfolio"
          >
            Portfolio
          </a>
        </li>
      </menu>
    </div>
  </nav>
</div>
`;
